create_jenkins_job.py

```diff
@@ -185,6 +185,20 @@ def create_job(os_name, job_name, template_file, additional_dict):
             'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
         })
 
+        # configure nightly job for testing with address sanitizer on linux
+        if os_name == 'linux':
+            asan_build_args = data['build_args_default'].replace('--cmake-args',
+                '--cmake-args -DOSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS=ON') + \
+                ' --mixin asan-gcc --packages-up-to test_communication'
```

Comment:
This looks problematic. If the `--cmake-args` in `build_args_default` are followed by actual CMake arguments after the replace they will be behind `--packages-up-to test_communication`.

Reply:
If I understand you correctly that should be handled.

1. Say `data['build_args_default']` is the following string, which includes --cmake-args
```python
'--event-handlers console_cohesion+ console_package_list+ --cmake-args -DINSTALL_EXAMPLES=OFF -DSECURITY=ON`'
```
2. `data['build_args_default'].replace('--cmake-args','--cmake-args -DOSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS=ON')` returns a new string with the inserted cmake arguments prepended to the list that was present previously:
```python
'--event-handlers console_cohesion+ console_package_list+ --cmake-args -DOSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS=ON -DINSTALL_EXAMPLES=OFF -DSECURITY=ON'
```
3. The new string has the remaining colcon arguments concatenated to the full list of cmake arguments yielding:
```python
'--event-handlers console_cohesion+ console_package_list+ --cmake-args -DOSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS=ON -DINSTALL_EXAMPLES=OFF -DSECURITY=ON --mixin asan-gcc --packages-up-to test_communication'
```

Reply:
I think I just missed the trailing `)` and didn't see that the next string starting with `--mixin` is independently concatenated. Sorry for the noise.

Reply:
No worries. It's a valid thing to check